node.hcl

```diff
@@ -6,6 +6,7 @@ strip = 1
 env = {
   "COREPACK_HOME": "${HERMIT_ENV}/.hermit/node",
   "NPM_CONFIG_PREFIX": "${HERMIT_ENV}/.hermit/node",
+  "NPM_CONFIG_CACHE": "${NPM_CONFIG_PREFIX}/cache",
```

Comment:
I'm not super familiar with the nuances of NPM caching. In general the approach we've taken is to put things that might conflict with other repositories into the `.hermit/<pkg>`, while things like shared caches are kept global under the assumption that they should already be robust enough. An example of this is Go's module cache, which is kept global.

 I'm happy to defer to your judgement on which side of this line NPM_CONFIG_CACHE falls under, WDYT?

Reply:
Here is more information on it: https://docs.npmjs.com/cli/v6/commands/npm-cache

I'm doing this in all of my projects that use Hermit, for two reasons:

1. Each hermit environment should be self contained, which is especially important for things like GitHub CI caches, where the entire `.hermit` folder can be cached between builds to save build minutes.
2. Helps when working offline/with no network, as that folder will contain HTTP responses that `npm` might make (`--prefer-offline`).

I suggested that change because I noticed that other toolchains like `rust.hcl` does this by default, which is helpful in the above scenarios.

However, it should be noted that It has the downside of an increased directory size per project (as with any machine/local cache). Up to you for the trade off.

Reply:
Aah interesting, I hadn't considered caching `.hermit` between builds. Not a bad idea!

Let's give it a shot and see how it goes.